src/Kestrel.Core/Internal/Http2/Http2FrameWriter.cs

```diff
@@ -50,6 +51,39 @@ public Task Write100ContinueAsync(int streamId)
             return Task.CompletedTask;
         }
 
+        public void WriteHeaders(int streamId, int statusCode, IHeaderDictionary headers)
+        {
+            lock (_writeLock)
```

Comment:
What is this lock protecting against? In the async version of this method, the lock is released before the write tasks complete.

Reply:
It protects access to `_outgoingFrame`.